### PR TITLE
Convert useSwapsEthToken hook to a selector

### PR DESCRIPTION
**Pull request by danjm**
The `useSwapsEthToken` does not need to be a hook, as it does not rely on other hooks (with the exception of useSelector, the need for which can just be met with a selector). It can be simplified to being a selector. This will also make it easier to use in more contexts (as it won't need to follow the rules of hooks)

### CODE REVIEW

File: ui/app/selectors/selectors.js [hunk 1]
```diff
@@ -11,6 +11,11 @@ import {
   checksumAddress,
   getAccountByAddress,
 } from '../helpers/utils/util';
+import {
```
Comment (brad-decker) [inline ui/app/selectors/selectors.js:14]: Review note: this is just the code from the hook, now as a selector, it doesn't appear to be modified except in name, and that it is now a selector that expects state as an argument.

File: ui/app/selectors/selectors.js [hunk 2]
```diff
@@ -381,3 +386,50 @@ export function getUSDConversionRate(state) {
...
+ * minimal token units (according to its decimals).
+ * `string` is the token balance in a readable format, ready for rendering.
+ *
+ * Swaps treats ETH as a token, and we use the ETH_SWAPS_TOKEN_OBJECT constant
+ * to set the standard properties for the token. The getSwapsEthToken selector
+ * extends that object with `balance` and `balance` values of the same type as
+ * in regular ERC-20 token objects, per the above description.
+ *
+ * @returns {SwapsEthToken} The token object representation of the currently
+ * selected account's ETH balance, as expected by the Swaps API.
```
Comment (brad-decker) [inline ui/app/selectors/selectors.js:419]: This doc block now should probably have a @param given it expects state.

**Review by brad-decker — CHANGES_REQUESTED** [review]
Is the hook file still being used after this? it doesn't appear to be removed in this PR.

**Review by ryanml — DISMISSED** [review]
++

**Comment by danjm** [comment]
@brad-decker your comments are addressed in the latest commit

**Review by koosy79 — APPROVED** [review]

**Review by brad-decker — APPROVED** [review]

**Pull request merged**